### ISSUE TITLE
neo4j-admin backup does not handle IPv6

### ISSUE THREAD

**Issue by mikernorton** [neo4j/neo4j#10792]
neo4j-admin backup gives an error when passed an IPv6 address in the "--from" argument.

- Neo4j version: neo4j-enterprise 3.2.8
- Operating system: RHEL 7.3 (VM)
- API/Driver: neo4j-admin

- **Steps to reproduce**
This was tested in a causal cluster.
-Bind the backup port to an IPv6 address, in my case, fd00:ce10::2 (This step may not be needed since the error happens early in the command parsing.)
-Generate a backup from the same machine using neo4j-admin:
**neo4j-admin backup --backup-dir=/dbdata/backup/db --name=graph.db --from=[fd00:ce10::2]:6362**

- Expected behavior
Backup proceeds with success!

- Actual behavior
Backup fails during command line parsing with error:
**for input string "ce10"**
followed by the backup usage.

I did try a number of different ways of adding in the address, including without brackets and a few even more exotic things.

**Comment by technige** [comment]
@mikernorton Did you try enclosing the entire address in quotes, i.e. `--from="[fd00:ce10::2]:6362"`?

**Comment by mikernorton** [comment]
Sorry, didn't see the notification. Yes, I tried putting quotes around it as well as not using the brackets. The causal_clustering.initial_discovery_members doesn't allow the brackets but does work, and most parameters like dbms.connector.http.listen_address seem to like having brackets to seperate IPv6 from port, as it would be in a url, so it is a bit of a soup.

**Comment by martinfurmanski** [comment]
No progress, but FYI we are tracking this internally and a resolution will come.

**Comment by mikernorton** [comment]
Any status on this? Last time I checked it was still an issue. It seems we'll be using IPv6 soon and this will be higher priority.

**Comment by mikernorton** [comment]
This still happens in 3.5.11, now the error says:
[root@neo2 /]# /usr/bin/neo4j-admin backup --backup-dir=/neo4jdb/backup/db --name=graph.db --from="[fd00:ce10:]:3"
...
unexpected error: hostname cannot contain '[' or ']'

I'll be escalating this soon through our account manager, as we aren't getting any progress.

**Comment by lutovich** [comment]
Hi @mikernorton,

Sorry for the delayed response!

I can reproduce the described problem. It is caused by an attempt to resolve a default backup address in case no `--from` is provided. Resolution logic reads `dbms.backup.address` setting from `neo4j.conf` and is not prepared to handle IPv6 addresses.

A temporary workaround would be:

```bash
$ echo "dbms.backup.address=127.0.0.1" > /tmp/backup-only-neo4j.conf
$ neo4j-admin backup --backup-dir=/dbdata/backup/db --name=graph.db --from=[fd00:ce10::2]:6362 --additional-config=/tmp/backup-only-neo4j.conf
```

It creates a temporary fake config file for the backup command. This config file overrides the backup address so that it is not IPv6. This override does not matter because the command contains an explicit `--from`. It simply makes the backup command parser happy.

This workaround shouldn't be used in a production environment. A proper fix should be available in the next 3.5 release. This issue will be updated when the fix is merged.

**Comment by lutovich** [comment]
Hi @mikernorton,

The fix has been merged into the private part of the 3.5 branch. It will be available in the next 3.5.13 release. Please feel free to update this issue if the problem persists.
